packages/grafana-ui/src/components/TimeSeries/utils.ts

```diff
@@ -296,7 +296,7 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<{
 
     let pointsFilter: uPlot.Series.Points.Filter = () => null;
 
-    if (customConfig.spanNulls !== true) {
+    if (customConfig.spanNulls !== true && showPoints !== VisibilityMode.Always) {
```

Comment:
not technically related, but avoids setting a wasteful filter when we always show all points.